recipes-bsp/device-tree/files/uz3eg-pciec/system-bsp.dtsi

```diff
@@ -1,186 +1,199 @@
-		compatible = "micron,m25p80"; /* 32MB */
-		#address-cells = <1>;
-		#size-cells = <1>;
-		reg = <0x0>;
-		spi-tx-bus-width = <1>;
-		spi-rx-bus-width = <4>; /* FIXME also DUAL configuration possible */
-		spi-max-frequency = <108000000>; /* Set to 108000000 Based on DC1 spec */
-	};
+   #address-cells = <1>;
+   #size-cells = <0>;
+   status = "okay";
+   is-dual = <1>; /* Set for dual-parallel QSPI config */
+   num-cs = <2>;
+   xlnx,fb-clk = <0x1>;
+   flash0: flash@0 {
+      /* The Flash described below doesn't match our board ("micron,n25qu256a"), but is needed */
+      /* so the Flash MTD partitions are correctly identified in /proc/mtd */
+      compatible = "micron,m25p80","jedec,spi-nor"; /* 32MB */
+      #address-cells = <1>;
+      #size-cells = <1>;
+      reg = <0x0>;
+      spi-tx-bus-width = <1>;
+      spi-rx-bus-width = <4>; /* FIXME also DUAL configuration possible */
+      spi-max-frequency = <108000000>; /* Set to 108000000 Based on DC1 spec */
+   };
 };
 
 /* SD0 eMMC, 8-bit wide data bus */
 &sdhci0 {
-	status = "okay";
-	bus-width = <8>;
-	max-frequency = <50000000>;
+   status = "okay";
+   bus-width = <8>;
+   max-frequency = <50000000>;
 };
 
 /* SD1 with level shifter */
 &sdhci1 {
-	status = "okay";
-	max-frequency = <50000000>;
-	no-1-8-v;	/* for 1.0 silicon */
+   status = "okay";
+   max-frequency = <50000000>;
+   no-1-8-v;   /* for 1.0 silicon */
 };
 
 &amba_pl {
-	// Remove definitions auto-generated by device tree generator
-	/delete-node/interrupt-controller@80030000; // axi_intc_0
-};
-
-&amba {
-	axi_intc_0: axi-interrupt-ctrl {
-		#interrupt-cells = <2>;
-		compatible = "xlnx,xps-intc-1.00.a";
-		interrupt-controller;
-		reg = <0x0 0x80030000 0x0 0x1000>;
-		xlnx,kind-of-intr = <0x0>;
-		xlnx,num-intr-inputs = <0x20>;
-		interrupt-parent = <&gic>;
-		// connected at pl_ps_irq0[5] (89+5=94) 
-		interrupts = <0 94 4>;
-	};
-
-	zyxclmm_drm {
-		compatible = "xlnx,zocl";
-		status = "okay";
-		interrupt-parent = <&axi_intc_0>;
-		interrupts = <0  4>, <1  4>, <2  4>, <3  4>,
-			<4  4>, <5  4>, <6  4>, <7  4>,
-			<8  4>, <9  4>, <10 4>, <11 4>,
-			<12 4>, <13 4>, <14 4>, <15 4>,
-			<16 4>, <17 4>, <18 4>, <19 4>,
-			<20 4>, <21 4>, <22 4>, <23 4>,
-			<24 4>, <25 4>, <26 4>, <27 4>,
-			<28 4>, <29 4>, <30 4>, <31 4>;
-	};
+   zyxclmm_drm {
+      compatible = "xlnx,zocl";
+      status = "okay";
+      interrupt-parent = <&axi_intc_0>;
+      interrupts = <0  4>, <1  4>, <2  4>, <3  4>,
+         <4  4>, <5  4>, <6  4>, <7  4>,
+         <8  4>, <9  4>, <10 4>, <11 4>,
+         <12 4>, <13 4>, <14 4>, <15 4>,
+         <16 4>, <17 4>, <18 4>, <19 4>,
+         <20 4>, <21 4>, <22 4>, <23 4>,
+         <24 4>, <25 4>, <26 4>, <27 4>,
+         <28 4>, <29 4>, <30 4>, <31 4>;
+   };
 };
 
 &psgtr {
-	/* sata, usb3, dp */
-	clocks = <&gtr_clk1>, <&gtr_clk2>, <&gtr_clk3>;
-	clock-names = "ref1", "ref2", "ref3";
+   /* NC, SATA, USB3, DP */
```

Comment:
Why is NC being added to the comment but there is not an additional clock added? i.e. why does the comment list 4 descriptions for 3 clocks? In other `psgtr` nodes there are 4 clocks, maybe the comment just got copy/pasted from one of those?

Reply:
The NC notes there is a No Connect.  There can be up to 4 GTR reference clocks described in the devicetree, but our hw platform only uses 3 and we note the 4th as NC.  This is what was recommended by the engineer at Xilinx.  The UZ7EV uses all 4 PS GTR reference clocks (though not connected the same to the Zynq MPSoC, and the 4th ref clock is used for PCIe).

Reply:
Ah, that makes sense